static/app/utils/profiling/flamegraph.ts

```diff
@@ -10,75 +10,72 @@ export class Flamegraph {
   profile: Profile;
   frames: FlamegraphFrame[] = [];
 
-  name: string;
   profileIndex: number;
-  startedAt: number;
-  endedAt: number;
 
   inverted?: boolean = false;
   leftHeavy?: boolean = false;
 
   depth = 0;
-  duration = 0;
   configSpace: Rect = new Rect(0, 0, 0, 0);
 
   formatter: (value: number) => string;
   frameIndex: Record<string, FlamegraphFrame> = {};
 
+  static Empty(): Flamegraph {
+    return new Flamegraph(Profile.Empty(), 0, {
+      inverted: false,
+      leftHeavy: false,
+    });
+  }
+
+  static From(from: Flamegraph, {inverted = false, leftHeavy = false}): Flamegraph {
+    return new Flamegraph(from.profile, from.profileIndex, {inverted, leftHeavy});
+  }
+
   constructor(
     profile: Profile,
     profileIndex: number,
-    {inverted = false, leftHeavy = false}: {inverted?: boolean; leftHeavy?: boolean} = {}
+    {inverted = false, leftHeavy = false}: {inverted?: boolean; leftHeavy?: boolean} = {},
+    configSpace?: Rect
   ) {
     this.inverted = inverted;
     this.leftHeavy = leftHeavy;
 
     // @TODO check if we can not keep a reference to the profile
     this.profile = profile;
-
-    this.duration = profile.duration;
     this.profileIndex = profileIndex;
-    this.name = profile.name;
-
-    this.startedAt = profile.startedAt;
-    this.endedAt = profile.endedAt;
 
+    // If a custom config space is provided, use it and draw the chart in it
     this.frames = leftHeavy
-      ? this.buildLeftHeavyGraph(profile)
-      : this.buildCallOrderGraph(profile);
+      ? this.buildLeftHeavyGraph(profile, configSpace ? this.profile.startedAt : 0)
+      : this.buildCallOrderGraph(profile, configSpace ? this.profile.startedAt : 0);
 
     this.formatter = makeFormatter(profile.unit);
 
-    if (this.duration) {
-      this.configSpace = new Rect(this.startedAt, 0, this.duration, this.depth);
-    } else {
-      // If the profile duration is 0, set the flamegraph duration
-      // to 1 second so we can render a placeholder grid
+    // If the profile duration is 0, set the flamegraph duration
+    // to 1 second so we can render a placeholder grid
+    this.configSpace = new Rect(
+      0,
+      0,
+      this.profile.unit === 'microseconds'
+        ? 1e6
+        : this.profile.unit === 'milliseconds'
+        ? 1e3
+        : 1,
+      this.depth
+    );
+
+    if (this.profile.duration) {
       this.configSpace = new Rect(
-        this.startedAt,
         0,
-        this.profile.unit === 'microseconds'
-          ? 1e6
-          : this.profile.unit === 'milliseconds'
-          ? 1e3
-          : 1,
+        0,
+        configSpace ? configSpace.right : this.profile.endedAt,
```

Comment:
```suggestion
        configSpace ? configSpace.width : this.profile.endedAt,
```

If you use `configSpace.right` here, this would be wrong if `configSpace.x` is non zero right?